islandora_pathauto.module

```diff
@@ -141,45 +141,108 @@ function islandora_pathauto_islandora_object_purged($pid) {
+      $unions .= 'UNION {
+              $object <fedora-model:label> $title ;
+              <fedora-model:hasModel> $cmodel ;
+              <fedora-model:state> <fedora-model:Active> .
+              FILTER(sameTerm($cmodel, <info:fedora/' . $contentmodel . '>))}
+              }';
+    }
+    $query = 'SELECT $object $title $cmodel
+          FROM <#ri>
+          WHERE { {
+              $object <fedora-model:label> $title ;
+              <fedora-model:hasModel> $cmodel ;
+              <fedora-model:state> <fedora-model:Active> .
+              FILTER(sameTerm($cmodel, <info:fedora/' . $first_contentmodel . '>))
+              }' . $unions . '
+              ORDER BY $title';
+    $tuque = islandora_get_tuque_connection();
+    if ($tuque) {
+      try {
+        $results = $tuque->repository->ri->query($query, 'sparql');
+        return $results;
+      }
+      catch (Exception $e) {
+        return array();
+      }
+    }
+    else {
+      IslandoraTuque::getError();
+    }
+  }
+  elseif ((module_exists('islandora_solr') == TRUE) && ($query_language == 'islandora_solr_query_backend')) {
+    module_load_include('inc', 'islandora_solr_search', 'includes/utilities');
+    $query_processor = new IslandoraSolrQueryProcessor();
+    $query_processor->solrQuery = 'RELS_EXT_hasModel_uri_s:(*' . str_replace(":", "\:", implode(" OR *", $models)) . ')';
+    $query_processor->solrStart = 0;
+    $query_processor->solrLimit = variable_get('islandora_pathauto_query_limit', '10000');
+    $query_processor->solrParams = array(
+      'fl' => "PID,fgs_label_mt,RELS_EXT_hasModel_uri_s",
+      'fq' => "",
+    );
+    $url = parse_url(variable_get('islandora_solr_url', 'localhost:8080/solr'));
+    $solr = new Apache_Solr_Service($url['host'], $url['port'], $url['path'] . '/');
+    $solr->setCreateDocuments(FALSE);
     try {
-      $results = $tuque->repository->ri->query($query, 'sparql');
-      $count = 0;
-      foreach ($results as $result) {
-        $pid = $result['object']['value'];
-        $object = islandora_object_load($pid);
-        $result = islandora_pathauto_create_alias($object, 'bulkupdate');
-        if ($result != '') {
-          $count += $result;
-        }
+      $results = $solr->search($query_processor->solrQuery, $query_processor->solrStart, $query_processor->solrLimit, $query_processor->solrParams, 'GET');
+      $tmp = json_decode($results->getRawResponse(), TRUE);
+      $results = array();
+      foreach ($tmp['response']['docs'] as $trip) {
+        $results[]['object']['value'] = $trip['PID'];
       }
-      drupal_set_message($count . ' islandora aliases were updated.');
+      return $results;
     }
     catch (Exception $e) {
-      if ($e->getCode() == '404') {
-        return FALSE;
-      }
-      else {
-        return NULL;
+      return array();
+    }
+  }
+}
+
+/**
+ * Implements hook_pathauto_bulkupdate().
+ */
+function islandora_pathauto_pathauto_bulkupdate() {
```

Comment:
You need to rebase your code because a lot has been added since Jun 2016. eg: https://github.com/Islandora/islandora_pathauto/blob/7.x/islandora_pathauto.module#L448